lib/ret.ex

```diff
@@ -1,158 +1,306 @@
-      reassign_avatar_listings(account_to_delete, acting_account)
-      reassign_parent_avatars(account_to_delete, acting_account)
-      reassign_scene_listings(account_to_delete, acting_account)
-
-      delete_entities_with_owned_files(
-        from(avatar in Avatar, where: avatar.account_id == ^account_to_delete.account_id),
-        Avatar.file_columns()
-      )
+  Returns `:ok` if successful, otherwise it returns `{:error, reason}`.
 
-      delete_entities_with_owned_files(
-        from(asset in Asset, where: asset.account_id == ^account_to_delete.account_id),
-        @asset_file_columns
-      )
+  Error reasons are:
 
-      project_query = from(project in Project, where: project.created_by_account_id == ^account_to_delete.account_id)
-      scene_query = from(scene in Scene, where: scene.account_id == ^account_to_delete.account_id)
+  * `:not_found` - an account could not be found with the given `id`
+  * `:forbidden` - the user does not have sufficient authorization
+  * `:failed` - the account data was unable to be deleted for an unknown reason
 
-      project_and_scene_owned_files =
-        collect_owned_files(project_query, @project_file_columns) ++
-          collect_owned_files(scene_query, @scene_file_columns)
+  """
+  @spec delete_account(Account.id(), Account.t()) :: :ok | {:error, :not_found | :forbidden | :failed}
+  def delete_account(id, %Account{} = user) when is_serial_id(id) do
+    account_to_delete = Repo.get(Account, id)
 
-      Repo.delete_all(project_query)
-      Repo.delete_all(scene_query)
+    cond do
+      !account_to_delete ->
+        {:error, :not_found}
 
-      delete_owned_files(project_and_scene_owned_files)
+      not can?(user, delete_account(account_to_delete)) ->
+        {:error, :forbidden}
 
-      delete_owned_files(Repo.all(from(o in OwnedFile, where: o.account_id == ^account_to_delete.account_id)))
+      true ->
+        reassignment = [set: [account_id: user.account_id]]
 
-      case Repo.delete(account_to_delete) do
-        {:ok, _} -> :ok
-        {:error, _} -> {:error, :failed}
-      end
-    else
-      {:error, :forbidden}
+        Ecto.Multi.new()
+        |> Ecto.Multi.update_all(
+          :reassign_avatar_listing_owned_files,
+          avatar_listing_owned_file_assignment_query(id),
+          reassignment
+        )
+        |> Ecto.Multi.update_all(
+          :reassign_avatar_listing_avatars,
+          avatar_listing_avatar_assignment_query(id),
+          reassignment
+        )
+        |> Ecto.Multi.update_all(
+          :reassign_avatar_listings,
+          avatar_listing_assignment_query(id),
+          reassignment
+        )
+        |> Ecto.Multi.update_all(
+          :reassign_avatar_owned_files,
+          avatar_owned_file_assignment_query(id),
+          reassignment
+        )
+        |> Ecto.Multi.update_all(
+          :reassign_parent_avatars,
+          parent_avatar_assignment_query(id),
+          reassignment
+        )
```

Comment:
I don't see a reassign_parent_scenes... Where is that handled for "remixed" project scenes?

Reply:
I see reassign_scenes but it doesnt seem to search for remixed scenes like remixed avatars. -- Maybe this isn't necessary by how the "remixing" scenes works. -- I saw a test case that checks this, but I'm not clear in the code where it happens.

Reply:
From what I recall Brian P telling me, parent scenes are different from parent avatars. Remixed scenes are copied, whereas remixed avatars are merely referenced. This means we can delete parent scenes, whereas we have to reassign parent avatars.

Reply:
Ah interesting. Did you poke around some production data to confirm that? seems slightly surprising we would copy them.

Reply:
@robinkwilson @netpro2k You two were absolutely correct. Pushing up a code fix now.